pkg/descheduler/evictions/evictions.go

```diff
@@ -90,11 +90,11 @@ func (pe *PodEvictor) EvictPod(pod *v1.Pod, node *v1.Node) (success bool, err er
 	success, err = EvictPod(pe.client, pod, pe.policyGroupVersion, pe.dryRun)
 	if success {
 		pe.nodepodCount[node]++
-		klog.V(1).Infof("Evicted pod: %#v (%#v)", pod.Name, err)
+		klog.V(1).Infof("Evicted pod: %#v in namespace %#v", pod.Name, pod.Namespace)
 		return success, nil
 	}
 	// err is used only for logging purposes
-	klog.Errorf("Error when evicting pod: %#v (%#v)", pod.Name, err)
+	klog.Errorf("Error evicting pod: %#v in namespace %#v (%#v)", pod.Name, pod.Namespace, err)
```

Comment:
If you're adding logs everywhere `EvictPod` is called to log the error, do you also need to log it here? I'd say we could get rid of this and keep the logs elsewhere

Reply:
`err` logged here is different from the error logged outside. If `EvictPod` returns non-nil error, this line is not executed.

Reply:
I think with your point in https://github.com/kubernetes-sigs/descheduler/pull/279/files#r424363568 though it might be redundant. So if we change `EvictPod` to just return an error, then `podEvictor.EvictPod` here can also be changed to just return an error right? No need for the bool anymore

Reply:
If the error is not nil, `EvictPod` signals the caller to stop evicting any pod. In our case it's due to reaching the maximum number of pods descheduler is allowed to evict per a node. Any other attempt to evict a pod from the same node will result in non-nil error. Thus, there's no reason to iterate through other pods. Whereas, in case of nil error, both `(true, nil)` and `(false, nil)` are valid cases. Negative success allows the strategy to react on the fact a pod was not evicted. In case it's not a blocker for a particular strategy, it may decide to continue evicting. Other strategies may decide to break and continue processing another node.

Reply:
Ah I see, in that case this lgtm. Thanks for clarifying @ingvagabund

Reply:
Sounds like no changes are required. Marking as resolved.